syntax = "proto3";

service DatanodeService {
  rpc WriteData(WriteRequest) returns (WriteResponse);
  rpc ReadData(ReadRequest) returns (ReadResponse);
}

message WriteRequest {
  string file_id = 1;
  bytes data = 2;
}

message WriteResponse {
  bool success = 1;
  bool fail = 2;
}

message ReadRequest {
  string file_id = 1;
  int64 offset = 2;
  int32 size = 3;
}

message ReadResponse {
  bytes data = 1;
  bool success = 2;
}
